modules/bluefish/consumer/bluefish_consumer.cpp

```diff
@@ -237,15 +455,85 @@ struct bluefish_consumer : boost::noncopyable
 		});
 	}
 
-	void display_frame(core::const_frame frame)
+	static void dma_present_thread_actual(void* arg)
```

Comment:
Was there ever a need for this argument to be `void*` instead of `bluefish_consumer*`? Anyway with the changes I suggested regarding the thread constructor this signature will be changed to `void dma_present_thread_actual()` with ordinary implicit `this` pointer.